### DISCUSSION TITLE
Оптимизация количества запросов к БД при использовании custom/my_meta.ini

### DISCUSSION THREAD

**Discussion by W1SE95** [maxsite/cms#483] [Q&A] [answered]
Здравствуйте. 
Подскажите пожалуйста, каким-то образом можно сократить количество запросов к БД при считывании мета ключей у записей внутри цикла?

`if ($pages) {
    foreach ($pages as $page) {  
	$show_loan = mso_get_meta('show_loan', 'page', $current_page_id, 'meta_value');
	$logo_loan = mso_get_meta('logo_loan', 'page', $current_page_id, 'meta_value');
... 
и таких запросов у каждого элемента цикла 15шт.
}
}
`

Когда на одной странице показывается всего лишь 7 записей получается SQL: 111, как по мне это очень много.
Есть вариант обращаться 1 запросом и брать все мета данные записи внутри цикла сразу?

![Screenshot_12](https://user-images.githubusercontent.com/60783381/151651677-3567fede-9ba9-42cd-819d-a7b50d368a00.jpg)

**Comment by maxsite** [comment]
Лучший вариант — это работать с мета как обычно, но результат их получения, или даже готовые функции — закэшировать. Так вы снизите sql-запросы до нуля для повторных посетителей.

**Comment by maxsite** [comment]
Да, и ещё момент. Не знаю как именно вы получаете записи, но в принципе все её мета-поля уже есть в массиве `$pages`.

**Reply by W1SE95** [reply]
![Screenshot_13](https://user-images.githubusercontent.com/60783381/151655161-ed55dc34-761a-4b56-a0d6-7b00ec406957.jpg)
Таким способом

**Reply by W1SE95** [reply]
![Screenshot_16](https://user-images.githubusercontent.com/60783381/151655437-18c4ec39-c57d-440a-b542-235c358f7651.jpg)
Вывел через print_R, получается нужно из [page_meta] выбирать нужное и все? Я правильно понял?

**Reply by W1SE95** [reply]
У меня получается таким образом, это правильно?
![Screenshot_17](https://user-images.githubusercontent.com/60783381/151656024-e5140d85-508b-4bf6-b7cd-358c70a45495.jpg)

**Comment by maxsite** [comment, marked as answer]
Просто сделайте `pr($pages);` там увидите массив метаданных страницы. Дальше работает с этим массивом в цикле.
